Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp

```diff
@@ -986,8 +988,10 @@ ThrowCompletionOr<DurationRecord> add_duration(VM& vm, double years1, double mon
         }
 
         // b. Let result be ? BalanceDuration(d1 + d2, h1 + h2, min1 + min2, s1 + s2, ms1 + ms2, mus1 + mus2, ns1 + ns2, largestUnit).
-        // FIXME: Narrowing conversion from 'double' to 'i64'
-        auto result = TRY(balance_duration(vm, days1 + days2, hours1 + hours2, minutes1 + minutes2, seconds1 + seconds2, milliseconds1 + milliseconds2, microseconds1 + microseconds2, Crypto::SignedBigInteger::create_from(nanoseconds1 + nanoseconds2), largest_unit));
+        VERIFY(trunc(nanoseconds1 + nanoseconds2) == nanoseconds1 + nanoseconds2);
+        if (!isfinite(nanoseconds1 + nanoseconds2))
+            return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidDuration);
```

Comment:
Adding this check here (and in 2 other places) is needed because I don't allow infinity and NaN in BigIntegers.

Perhaps we could use a variant of BigInteger here that just throws RangeError on infinity however I don't know enough about Temporal to know whether that would be correct and wanted.

These are the only ones I found by test262 and when replacing.